### ISSUE TITLE
I have to restart arangod server every time I made changes

### ISSUE THREAD

**Issue by ilearnio** [arangodb/arangodb#1927]
It seems that there was no such bug in ArangoDB v2.8, but after I upgraded to v3.0.1 it's now a pain to develop my Foxx service. Now every time I make some change to my code I have to restart arangod otherwise it may not take effect. Sometimes it works, sometimes not.

I'll better show it:
http://take.ms/Q0u0M

As you can see it works very strangely. I first thought that it was a problem with `graphql-sync` plugin because most of the times it occurs when I'm updating files of some GraphQL queries, but as you can see at the end of the video I have captured when the same thing happened when I updated test file. Am I doing something wrong or it seems like a bug?

Thank you

**Comment by dothebart** [comment]
hm, I'm currently trying to reproduce this by editing the itzpapalotl service, and fail to reproduce this on linux and mac. I need to try the downloaded mac version. Can you revalidate whether the timestamp of the file changes?

**Comment by ilearnio** [comment]
@dothebart I just watched the recorded video again and realized that even in the end of the video there was no any problems with my test files themselves (`...specs.js`) but only when editing files of GraphQL queries. So it's definitely has something to do with `graphql-sync` plugin. Maybe arangodb caches output from functions of npm packages somehow.

**Comment by ilearnio** [comment]
I just tried to do the same thing with `console.log`s in `node_modules/graphql-sync/lib/graphal.js:43` and the same thing happens. When I save this file and run tests it also not always output to the console with new messages, but with old ones instead. Strange, the code of the package looks pretty simple

**Comment by ilearnio** [comment]
I was able to reproduce it in a new foxx app. I created a repo for you to test it https://github.com/ilearnio/arangodb-bug .
Try to add this app, then run tests, it will pass. Then try to change the value in `main.js:26` from 1 to something else, it will not pass, then change it again to something random it will pass (as if there was 1)! 

Also to catch this bug you can directly change line 7 in `node_modules/graphql-sync/lib/index.js` from `graphql: require('./graphql').graphql,` to something like `graphql: () => console.log(1),`, start the server, then change 1 to 2 and it will still output 1

**Comment by dothebart** [comment]
ok, can reproduce on linux. Once main.js:26 is edited, it won't flip back.

**Comment by pluma** [comment]
I think this is a duplicate of the same underlying problem as #1892.

**Comment by ilearnio** [comment]
@pluma Looks like so. But what I realized today is that I have this issue even without tests. You can reproduce it with this simple example in `main.js`:

```js
const createRouter = require('@arangodb/foxx/router')()

router.get('/foo', function(req, res) {
  res.json({ foo: 'bar' })
})

module.context.use(router)
```

Try to execute /foo in the browser, then update 'bar' to something else, then refresh the page. The value will be the same

**Comment by baslr** [comment]
@ilearnio can you check if this is still an issue with 3.1? Thanks in advance.

**Comment by ilearnio** [comment]
@baslr This issue was fixed in 3.0.x, so I guess 3.1 should also be OK.

**Comment by baslr** [comment]
@ilearnio thanks for reporting back 🦊
